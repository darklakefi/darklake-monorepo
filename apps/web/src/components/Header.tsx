import Link from "next/link";
import Image from "next/image";
import ConnectedWalletDropdown from "./ConnectedWalletDropdown";

const Header = () => {
  // TODO: wire with added/connected wallet
  const currentWalletAddress = null;
  return (
    <header className="flex flex-row justify-between items-center pb-[64px]">
      <Link href="/" title="Darklake.fi" className="active:opacity-80">
        <Image src="/images/logo-h-darklake.png" alt="darklage logo" height={24} width={147} />
      </Link>
      {currentWalletAddress ? <ConnectedWalletDropdown currentWalletAddress={currentWalletAddress} /> : null}
    </header>
  );
};

export default Header;
