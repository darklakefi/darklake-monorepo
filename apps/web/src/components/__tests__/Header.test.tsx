import { act, render, RenderResult } from "@testing-library/react";
import Header from "@/components/Header";

describe("<Header />", () => {
  it("renders successfully", async () => {
    let baseElement: RenderResult["baseElement"] | undefined;

    await act(async () => {
      ({ baseElement } = render(<Header />));
    });

    expect(baseElement).toMatchSnapshot();
  });
});
