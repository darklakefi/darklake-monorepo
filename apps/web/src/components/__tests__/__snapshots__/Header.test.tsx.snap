// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders successfully 1`] = `
<body>
  <div>
    <header
      class="flex flex-row justify-between items-center z-20 mb-20"
    >
      <div
        class="flex items-center justify-between gap-x-[11px]"
      >
        <a
          class="active:opacity-80"
          href="/"
          title="Darklake.fi"
        >
          <img
            alt="darklake logo"
            data-nimg="1"
            decoding="async"
            height="24"
            loading="lazy"
            src="/images/logo-h-darklake-beta.svg"
            style="color: transparent;"
            width="213"
          />
        </a>
      </div>
    </header>
  </div>
</body>
`;
